lightly/models/modules/masked_vision_transformer_timm.py

```diff
@@ -0,0 +1,169 @@
+        mask_token: Optional[Parameter] = None,
+    ) -> None:
+        super().__init__()
+        self.vit = vit
+        self.mask_token = (
+            mask_token
+            if mask_token is not None
+            else Parameter(torch.zeros(1, 1, self.vit.embed_dim))
+        )
+        self._initialize_weights()
+
+    @property
+    def sequence_length(self) -> int:
+        seq_len: int = self.vit.patch_embed.num_patches + self.vit.num_prefix_tokens
+        return seq_len
+
+    def forward(
+        self,
+        images: Tensor,
+        idx_mask: Optional[Tensor] = None,
+        idx_keep: Optional[Tensor] = None,
+    ) -> Tensor:
+        x = self.encode(images, idx_mask=idx_mask, idx_keep=idx_keep)
+        if self.vit.attn_pool is not None:
+            x = self.vit.attn_pool(x)
+        elif self.vit.global_pool == "avg":
+            x = x[:, self.vit.num_prefix_tokens :].mean(dim=1)
+        elif self.vit.global_pool:
+            x = x[:, 0]  # class token
+        return x
+
+    def encode(
+        self,
+        images: torch.Tensor,
+        idx_mask: Optional[torch.Tensor] = None,
+        idx_keep: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        # convert images to tokens
+        input = self.images_to_tokens(images)
+        # add prefix tokens if needed
+        input = self.add_prefix_tokens(input)
+
+        if idx_mask is not None:
+            input = utils.mask_at_index(input, idx_mask, self.mask_token)
+        # add positional encoding
+        input = self.add_pos_embed(input)
+
+        if idx_keep is not None:
+            input = utils.get_at_index(input, idx_keep)
+        # normalization layer
+        input = self.vit.norm_pre(input)
+        # apply Transformer blocks
+        input = self.vit.blocks(input)
+        # normalize
+        out: torch.Tensor = self.vit.norm(input)
+        return out
+
+    def images_to_tokens(self, images: Tensor) -> Tensor:
+        tokens: Tensor = self.vit.patch_embed(images)
+        if self.vit.dynamic_img_size:
+            tokens = tokens.permute(0, 3, 1, 2)  # NHWC -> NCHW
+            tokens = tokens.flatten(2).transpose(1, 2)  # NCHW -> NLC
+        return tokens
+
+    def add_prefix_tokens(self, x: Tensor, prepend_class_token: bool = True) -> Tensor:
+        prefix_tokens = []
+        if self.vit.cls_token is not None:
+            prefix_tokens.append(self.vit.cls_token.expand(x.shape[0], -1, -1))
+        if self.vit.reg_token is not None:
+            prefix_tokens.append(self.vit.reg_token.expand(x.shape[0], -1, -1))
+        if prefix_tokens:
+            x = torch.cat(prefix_tokens + [x], dim=1)
+        return x
+
+    def add_pos_embed(self, x: Tensor) -> Tensor:
+        x_prefix = (
+            x[:, : self.vit.num_prefix_tokens, :]
+            if self.vit.num_prefix_tokens
+            else torch.empty()
+        )
+        x = x[:, self.vit.num_prefix_tokens :, :] if self.vit.num_prefix_tokens else x
```

Comment:
```suggestion
        x_prefix = x[:, : self.vit.num_prefix_tokens, :]
        x = x[:, self.vit.num_prefix_tokens :, :]
```

This is simpler and does not create any overhead. Note that using `torch.empty()` here is dangerous because it will potentially create a tensor on the wrong device and with the wrong datatype. If you really need a new tensor then `x.new_empty(size)` should be used instead. This makes sure that the new tensor is on the correct device with the correct datatype.